### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by Sashk52**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>
```
Comment (mfedenko09) [inline pom.xml]:
```suggestion
</project>

```
add empty in end of project please

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (int i = 0; i < manufacturers.size(); i++) {
+            if (manufacturers.get(i).getId().equals(id)) {
+                Storage.manufacturers.remove(i);
+                return true;
+            }
+        }
+        return false;
+    }
+}
```
Comment (mfedenko09) [inline src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]:
```suggestion
        return Storage.manufacturers.remove(id);
    }
}
```

**Review by mfedenko09 — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java (hunk 2 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: done

File: pom.xml (hunk 1 above)
Comment (Sashk52) [reply pom.xml]: done

**Review by mfedenko09 — APPROVED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (Sofasmile) [inline .travis.yml]: let's add empty line at the end of the file

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
+        <maven.checkstyle.plugin.configLocation>
+            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
+
+    <dependencies>
+        <dependency>
```
Comment (Sofasmile) [inline pom.xml]: do we need this dependency?

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
```
Comment (Sofasmile) [inline src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: try to use stream api here

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (Manufacturer each : manufacturers) {
+            if (each.getId().equals(manufacturer.getId())) {
+                Storage.manufacturers.set(Math.toIntExact(each.getId() - 1L), manufacturer);
```
Comment (Sofasmile) [inline src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: is `id` the same as an element position in the list?

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        }
+        return null;
```
Comment (Sofasmile) [inline src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: return null is a bad practice, let's throw exception instead

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return Storage.manufacturers.remove(id);
```
Comment (Sofasmile) [inline src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: does this method work correctly?

File: src/main/java/taxi/model/Car.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+public class Car {
+    private Long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(String model, Manufacturer manufacturer) {
+        this.model = model;
+        this.manufacturer = manufacturer;
+        this.drivers = Collections.emptyList();
```
Comment (Sofasmile) [inline src/main/java/taxi/model/Car.java]:
```suggestion
        drivers = new ArrayList();
```

File: src/main/java/taxi/service/ManufacturerServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        return manufacturerDao.create(manufacturer);
+    }
+
+    @Override
+    public Manufacturer get(Long id) {
+        return Optional.of(Storage.manufacturers
+                .get(Math.toIntExact(id))).get();
```
Comment (Sofasmile) [inline src/main/java/taxi/service/ManufacturerServiceImpl.java]: we can't use Storage at service layer, let's use manufacturerDao instead

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java (hunk 8 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: No, Made correction.

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java (hunk 7 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: done

File: .travis.yml (hunk 3 above)
Comment (Sashk52) [reply .travis.yml]: done

File: pom.xml (hunk 4 above)
Comment (Sashk52) [reply pom.xml]: We don't use junit, so no.

File: src/main/java/taxi/model/Car.java (hunk 9 above)
Comment (Sashk52) [reply src/main/java/taxi/model/Car.java]: done

File: src/main/java/taxi/service/ManufacturerServiceImpl.java (hunk 10 above)
Comment (Sashk52) [reply src/main/java/taxi/service/ManufacturerServiceImpl.java]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java (hunk 5 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java (hunk 6 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: yes

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java [hunk 11]
```diff
@@ -18,13 +18,7 @@ public Manufacturer create(Manufacturer manufacturer) {
...
     @Override
     public Optional<Manufacturer> get(Long id) {
-        List<Manufacturer> manufacturers = Storage.manufacturers;
-        for (Manufacturer manufacturer : manufacturers) {
-            if (manufacturer.getId().equals(id)) {
-                Optional.of(manufacturer);
-            }
-        }
-        return Optional.empty();
+        return Storage.manufacturers.stream().filter(x -> x.getId().equals(id)).findFirst();
```
Comment (Sofasmile) [inline src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]:
```suggestion
        return Storage.manufacturers.stream()
                .filter(x -> x.getId().equals(id))
                .findFirst();
```

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java [hunk 12]
```diff
@@ -34,17 +28,23 @@ public Manufacturer create(Manufacturer manufacturer) {
 
     @Override
     public Manufacturer update(Manufacturer manufacturer) {
-        List<Manufacturer> manufacturers = Storage.manufacturers;
-        for (Manufacturer each : manufacturers) {
-            if (each.getId().equals(manufacturer.getId())) {
-                Storage.manufacturers.set(Math.toIntExact(each.getId() - 1L), manufacturer);
+        for (int i = 0; i < Storage.manufacturers.size(); i++) {
+            if (manufacturer.getId().equals(Storage.manufacturers.get(i).getId())) {
+                Storage.manufacturers.set(i, manufacturer);
```
Comment (Sofasmile) [inline src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java:35]:
```suggestion
                Storage.manufacturers.set(i, manufacturer);
                break;
```

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java [hunk 13]
```diff
@@ -34,17 +28,23 @@ public Manufacturer create(Manufacturer manufacturer) {
...
     public boolean delete(Long id) {
-        return Storage.manufacturers.remove(id);
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (int i = 0; i < manufacturers.size(); i++) {
+            if (manufacturers.get(i).getId().equals(id)) {
+                Storage.manufacturers.remove(i);
+                return true;
+            }
+        }
+        return false;
```
Comment (Sofasmile) [inline src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]:
```suggestion
        return Storage.manufacturers.removeIf(m -> m.getId().equals(id));
```

File: src/main/java/taxi/service/ManufacturerServiceImpl.java [hunk 14]
```diff
@@ -21,8 +20,7 @@ public Manufacturer create(Manufacturer manufacturer) {
 
     @Override
     public Manufacturer get(Long id) {
-        return Optional.of(Storage.manufacturers
-                .get(Math.toIntExact(id))).get();
+        return Optional.of(manufacturerDao.get(id)).get().get();
```
Comment (Sofasmile) [inline src/main/java/taxi/service/ManufacturerServiceImpl.java]:
```suggestion
        return manufacturerDao.get(id).get();
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java (hunk 11 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java (hunk 12 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java:35]: done

File: src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java (hunk 13 above)
Comment (Sashk52) [reply src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java]: done

File: src/main/java/taxi/service/ManufacturerServiceImpl.java (hunk 14 above)
Comment (Sashk52) [reply src/main/java/taxi/service/ManufacturerServiceImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**